packages/tradershub/src/features/cfd/screens/CreatePassword/CreatePassword.tsx

```diff
@@ -1,58 +1,49 @@
-import { TPlatforms } from '@/types';
-import { validPassword } from '@/utils';
-import { PlatformDetails } from '@cfd/constants';
-import { Button, PasswordInput, Text, useDevice } from '@deriv-com/ui';
+import React, { ChangeEvent } from 'react';
+import MT5PasswordIcon from '@/assets/svgs/ic-mt5-password.svg';
+import { useCFDContext } from '@/providers';
+import { Category, CFDPlatforms, PlatformDetails } from '@cfd/constants';
+import { useActiveTradingAccount } from '@deriv/api-v2';
+import { Modal, PasswordInput, Text, useDevice } from '@deriv-com/ui';
+import MT5PasswordFooter from '../../modals/MT5PasswordModal/MT5PasswordFooter';
 
 type TCreatePasswordProps = {
-    icon: ReactNode;
-    isLoading?: boolean;
     onPasswordChange?: (e: ChangeEvent<HTMLInputElement>) => void;
-    onPrimaryClick: () => void;
     password: string;
-    platform: TPlatforms.All;
 };
 
 /**
  * Component to create a password for the platform
- * @param icon
- * @param isLoading
  * @param onPasswordChange
- * @param onPrimaryClick
  * @param password
- * @param platform MT5 or Deriv X
- * @returns
  */
-const CreatePassword = ({
-    icon,
-    isLoading,
-    onPasswordChange,
-    onPrimaryClick,
-    password,
-    platform,
-}: TCreatePasswordProps) => {
+const CreatePassword = ({ onPasswordChange, password }: TCreatePasswordProps) => {
     const { isDesktop } = useDevice();
+    const { data: activeTrading } = useActiveTradingAccount();
+    const { cfdState } = useCFDContext();
 
+    const isDemo = activeTrading?.is_virtual;
+    const { platform = CFDPlatforms.MT5 } = cfdState;
     const { title } = PlatformDetails[platform];
+
     return (
-        <div className='inline-flex flex-col items-center w-full gap-24 rounded-default bg-system-light-primary-background lg:w-[360px]'>
-            {isDesktop && icon}
-            <div className='flex flex-col items-center justify-center text-center lg:gap-8'>
-                <Text weight='bold'>Create a {title} password</Text>
-                <Text size='sm'>You can use this password for all your {title} accounts.</Text>
-            </div>
-            <PasswordInput isFullWidth label={`${title} password`} onChange={onPasswordChange} value={password} />
-            {isDesktop && (
-                <Button
-                    disabled={!password || isLoading || !validPassword(password)}
-                    isLoading={isLoading}
-                    onClick={onPrimaryClick}
-                    size='lg'
-                >
-                    {`Create ${title} password`}
-                </Button>
-            )}
-        </div>
+        <React.Fragment>
+            <Modal.Header>
+                <Text weight='bold'>{`Create a ${isDemo ? Category.DEMO : Category.REAL} ${
+                    PlatformDetails.mt5.title
+                } account`}</Text>
+            </Modal.Header>
+            <Modal.Body>
+                {isDesktop && <MT5PasswordIcon />}
```

Comment:
I would rather have you wrap this in a div and put display= inline-block for large screens and display=none for anything that is smaller than it. Controlling with css is much more faster than using js 😉